shared/search/providers.ts

```diff
@@ -249,23 +238,154 @@ function getFileContent(
     return getFileContentFromApi(repo, commit, path)
 }
 
+/**
+ * Perform a search query for definitions. Returns results converted to locations,
+ * filtered by the language's definition filter, and sorted by proximity to the
+ * current text document path.
+ *
+ * @param args Parameter bag.
+ */
+async function searchAndFilterDefinitions({
+    doc,
+    repo,
+    path,
+    text,
+    filterDefinitions,
+    query,
+}: {
+    /** The current text document. */
+    doc: sourcegraph.TextDocument
+    /** The repository containing the current text document. */
+    repo: string
+    /** The path of the current text document. */
+    path: string
+    /** The content of the current text document */
+    text: string
+    /** The function used to filter definitions. */
+    filterDefinitions: FilterDefinitions
+    /** The search query. */
+    query: string
+}): Promise<sourcegraph.Location[]> {
+    // Perform search and perform pre-filtering before passing it
+    // off to the language spec for the proper filtering pass.
+    const searchResults = await search(query)
+    const preFilteredResults = searchResults.filter(result =>
+        isSymbolVisible(doc, path, result)
+    )
+
+    // Filter results based on language spec
+    const filteredResults = filterDefinitions(preFilteredResults, {
+        repo,
+        filePath: path,
+        fileContent: text,
+    })
+
+    return sortByProximity(
+        filteredResults.map(resultToLocation),
+        new URL(doc.uri)
+    )
+}
+
+/**
+ * Perform a search query for references. Returns results converted to locations.
+ * Results are not sorted in any meaningful way as these results are meant to be
+ * merged with other search query results.
+ *
+ * @param args Parameter bag.
+ */
+async function searchReferences({
+    query,
```

Comment:
This seems like a lot of ceremony to pass in a query string. Given this isn't public, can we just make this `async function searchReferences(query: string): Promise<sourcegraph.Location[]> {` ?

(I get extensibility, but we own all the call sites)

Reply:
We can't. It was written this way so that it could be passed to `searchWithFallback`, which has a generic input parameter extending `{query: string}`.